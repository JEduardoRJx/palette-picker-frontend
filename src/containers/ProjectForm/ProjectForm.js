import React, { Component } from 'react';
import './ProjectForm.scss'

export class ProjectForm extends Component {
  constructor() {
    super()
    this.state = {
      currentProject: {
        name: ''
      },
      allProjects: [],
      error: ''
    }
  }
  render() {
    return (
      <section className="project-details">
        <form className="project-form">
         <input type="text" name="projectName" placeholder="Enter Project Name"></input>
        </form>
        <select className="project-select">
          <option value="Project Name"> Project Name </option>
          <option value="Project Name 1"> Project 1 </option>
          <option value="Project Name 2"> Project 2 </option>
        </select>
      </section>
    )
  }
}


export default ProjectForm;