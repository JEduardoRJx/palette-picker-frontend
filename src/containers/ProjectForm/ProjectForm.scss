@import "../../variables.scss";

.project-details {
  width: 33.3%;
  text-align: left;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.project-form {
  input {
    @include base-input;
    font-style: italic;
  }
}

.project-select {
  @include base-input;
  margin-bottom: 10px;
}

.current-project {
  font-size: 3em;
  margin-top: 5%;
}

.view-all-projects-btn {
  background-color: #c63166;
  border-radius: 25px;
  box-shadow: 0 9px #999;
  color: #fff;
  cursor: pointer;
  font-size: 1.5em;
  height: 50px;
  margin: 15px;
  text-align: center;
  width: 300px;
  padding: 5px;
  &:hover { 
    background-color: #8f244a;
    font-style: italic;
  }
  &:active {
    background-color: #8f244a;
    font-style: italic;
    box-shadow: 0 5px #666;
    transform: translateY(4px);
  }
}
