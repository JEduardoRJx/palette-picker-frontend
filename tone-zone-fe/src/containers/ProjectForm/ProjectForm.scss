@import "../../variables.scss";

.project-details {
  width: 33.3%;
  text-align: left;
}

.project-form {
  input {
    @include base-input;
    font-style: italic;
  }
}

.project-select {
  @include base-input;
}