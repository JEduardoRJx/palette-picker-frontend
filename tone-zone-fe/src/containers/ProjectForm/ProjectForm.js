import React, { Component } from 'react';
import './ProjectForm.scss'

export class ProjectForm extends Component {
  constructor() {
    super()
    this.state = {
      currentProject: {
        name: ''
      },
      allProjects: [],
      error: ''
    }
  }
  render() {
    return (
      <section className="project-details">
        <form className="project-form">
          Enter Project Name: <input type="text" name="projectName"></input>
          <input type="submit"></input>
        </form>
        <select>
          <option value="Project Name"> Project Name </option>
          <option value="Project Name 1"> Project 1 </option>
          <option value="Project Name 2"> Project 2 </option>
        </select>
      </section>
    )
  }
}


export default ProjectForm;