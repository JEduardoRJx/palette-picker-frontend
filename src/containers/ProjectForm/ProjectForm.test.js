import React from 'react';
import { shallow } from 'enzyme';
import { ProjectForm } from './ProjectForm'

describe('ProjectForm ', () => {
  const mockProjects = [
    {
      id: 3,
      project_name: "spring",
      user_id: 1,
      created_at: "2019-12-10T04:14:29.628Z",
      updated_at: "2019-12-10T04:14:29.628Z"
    }
  ]

  it('should match the snapshot ', () => {
    const wrapper = shallow(<ProjectForm 
      projects={mockProjects}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});