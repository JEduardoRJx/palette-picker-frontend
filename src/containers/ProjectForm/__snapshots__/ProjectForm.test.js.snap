// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm  should match the snapshot  1`] = `
<section
  className="project-details"
>
  <form
    className="project-form"
  >
    <input
      name="currentProject"
      onChange={[Function]}
      placeholder="Enter New Project Name"
      type="text"
    />
  </form>
  <select
    className="project-select"
    onChange={[Function]}
  >
    <option
      value="Project Name"
    >
       Select existing Project 
    </option>
    <option
      key="3"
      value="spring"
    >
       
      spring
        
    </option>
  </select>
  <NavLink
    className="view-allprojects-btn"
    to="/projects"
  >
    View All Projects
  </NavLink>
</section>
`;
